lib/phoenix/channel.ex

```diff
@@ -124,17 +124,34 @@ defmodule Phoenix.Channel do
                                                                      {:leave, Socket.t} |
                                                                      {:error, reason :: term, Socket.t}
 
-  defmacro __using__(_options) do
+  defmacro __using__(options \\ []) do
     quote do
       @behaviour unquote(__MODULE__)
-      import unquote(__MODULE__)
+      @pubsub_server unquote(options[:pubsub_server])
```

Comment:
Also, you should unquote the options and access the value, otherwise you are assuming the options are a keyword list at compilation time (usually true but not always :)).